README.md

```diff
@@ -47,50 +49,81 @@ flow and returns the access token, refresh token and access token expiry date wh
-- **redirectUrl** - (`string`) *REQUIRED* the url that links back to your app with the auth code
-- **additionalParameters** - (`object` | `null`) additional parameters that will be passed in the authorization request.
-Must be string values! E.g. setting `additionalParameters: { hello: 'world', foo: 'bar' }` would add
-`hello=world&foo=bar` to the authorization request.
+* **issuer** - (`string`) _REQUIRED_ the url of the auth server
+* **clientId** - (`string`) _REQUIRED_ your client id on the auth server
+* **redirectUrl** - (`string`) _REQUIRED_ the url that links back to your app with the auth code
+* **scopes** - (`array<string>`) _REQUIRED_ the scopes for your token, e.g. `['email', 'offline_access']`
+* **additionalParameters** - (`object`) additional parameters that will be passed in the authorization request.
+  Must be string values! E.g. setting `additionalParameters: { hello: 'world', foo: 'bar' }` would add
+  `hello=world&foo=bar` to the authorization request.
 
-### `result`
+#### result
 
 This is the result from the auth server
-- **accessToken** - (`string`) the access token
-- **accessTokenExpirationDate** - (`string`) the token expiration date
-- **additionalParameters** - (`Object`) additional url parameters from the auth server
-- **idToken** - (`string`) the id token
-- **refreshToken** - (`string`) the refresh token
-- **tokenType** - (`string`) the token type, e.g. Bearer
+
+* **accessToken** - (`string`) the access token
+* **accessTokenExpirationDate** - (`string`) the token expiration date
+* **additionalParameters** - (`Object`) additional url parameters from the auth server
+* **idToken** - (`string`) the id token
+* **refreshToken** - (`string`) the refresh token
+* **tokenType** - (`string`) the token type, e.g. Bearer
 
 ### `refresh`
 
 This method will refresh the accessToken using the refreshToken. Some auth providers will also give
 you a new refreshToken
 
 ```js
-const result = await appAuth.refresh(refreshToken, scopes);
-// returns accessToken, accessTokenExpirationDate and (maybe) refreshToken
+import { refresh } from 'react-native-app-auth';
+
+const config = {
+  issuer: '<YOUR_ISSUER_URL>',
+  clientId: '<YOUR_CLIENT_ID>',
+  redirectUrl: '<YOUR_REDIRECT_URL>',
+  scopes: '<YOUR_SCOPES_ARRAY>',
+};
+
+const result = await refresh({
+  ...config,
+  refreshToken: `<REFRESH_TOKEN>`,
+});
 ```
 
-### `revokeToken`
+### `revoke`
 
 This method will revoke a token. The tokenToRevoke can be either an accessToken or a refreshToken
 
 ```js
-// note, sendClientId=true will only be required when using IdentityServer
-const result = await appAuth.revokeToken(tokenToRevoke, sendClientId);
+import { revoke } from 'react-native-app-auth';
+
+const config = {
+  issuer: '<YOUR_ISSUER_URL>',
+  clientId: '<YOUR_CLIENT_ID>',
+  redirectUrl: '<YOUR_REDIRECT_URL>',
+  scopes: '<YOUR_SCOPES_ARRAY>',
+};
+
+const result = await revoke({
+  ...config,
+  tokenToRevoke: `<TOKEN_TO_REVOKE>`
```

Comment:
I'm not sure whether this is a sensible API design, as it's not part of a config.

`config` is currently pretty agnostic to how it's used, but `tokenToRevoke` is a one off parameter. Since `revoke` will always affect a single token maybe the functions should be changed to either:

`revoke: (config: {| ... |}, token: string) => Promise<>`

or:

`revoke: (config: {| ... |}) => (token: string) => Promise<>`

same could apply to the `refreshToken` above, since it changes when calling `authorize`. I'm not sure what `authorize: (...) => Promise<T>` resolves to. Maybe `revoke` and `refresh` should just accept the result of `authorize`?

Reply:
I think the first suggestion makes sense, but with the second argument also being an object:

```js
revoke: (config: {| ... |}, { token: string }) => Promise<>
```